src/tcp.rs

```diff
@@ -37,7 +37,9 @@ impl TcpSocket {
 
         self.socket.set_interrupt_mask(
             bus,
-            socketn::Interrupt::SendOk as u8 & socketn::Interrupt::Timeout as u8,
+            socketn::Interrupt::SendOk as u8
```

Comment:
It seems like interrupts should be configured when creating the socket. We don't want to blindly enable them for everything in my opinion